src/EditorFeatures/Core.Wpf/NavigateTo/NavigateToItemProvider.Callback.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.NavigateTo;
+using Microsoft.CodeAnalysis.Text.Shared.Extensions;
+using Microsoft.VisualStudio.Language.NavigateTo.Interfaces;
+using Microsoft.VisualStudio.Text.PatternMatching;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.NavigateTo
+{
+    internal partial class NavigateToItemProvider
+    {
+        private class VisualStudioNavigateToCallback : INavigateToSearchCallback
+        {
+            private readonly INavigateToItemDisplayFactory _displayFactory;
+            private readonly INavigateToCallback _callback;
+
+            public VisualStudioNavigateToCallback(INavigateToItemDisplayFactory displayFactory, INavigateToCallback callback)
+            {
+                _displayFactory = displayFactory;
+                _callback = callback;
+            }
+
+            public void Done(bool isFullyLoaded = true)
```

Comment:
i would prefer we not have these optional arguments in our internal APIs.  it's too easy forthe caller to miss and reviewing does not make that omission obivous